src/core/operations/YaraRules.mjs

```diff
@@ -0,0 +1,82 @@
+/**
+ * @author Matt C [matt@artemisbot.uk]
+ * @copyright Crown Copyright 2019
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation";
+import OperationError from "../errors/OperationError";
+import Yara from "libyara-wasm";
+
+/**
+ * Yara Rules operation
+ */
+class YaraRules extends Operation {
+
+    /**
+     * YaraRules constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Yara Rules";
+        this.module = "Yara";
+        this.description = "Yara support";
+        this.infoURL = "https://en.wikipedia.org/wiki/YARA";
+        this.inputType = "ArrayBuffer";
+        this.outputType = "string";
+        this.args = [{
+            name: "Rules",
+            type: "code",
```

Comment:
I think the `text` type works here. No need to create another textarea argument unless you want it to be significantly different.

Reply:
It would, yes. Ideally it would be a drag and drop interface like over the input. I guess a browse button would make sense too.